app/models/agents/http_status_agent.rb

```diff
@@ -84,11 +85,11 @@ def check_this_url(url, local_headers)
 
       # Deal with failures
       if measured_result.result
-        final_url = boolify(interpolated['disable_redirect_follow']) ? url : measured_result.result.to_hash[:url]
+        final_url = boolify(interpolated['disable_redirect_follow']) ? url : measured_result.result.env.url.to_s
         payload.merge!({ 'final_url' => final_url, 'redirected' => (url != final_url), 'response_received' => true, 'status' => current_status })
         # Deal with headers
         if local_headers.present?
-          header_results = measured_result.result.headers.select {|header, value| local_headers.include?(header)}
+          header_results = measured_result.result.headers.slice(*local_headers)
```

Comment:
`local_headers` is a Ruby array, so the original code was case-sensitive anyway.